### PR TITLE
NLIC-2342 log4j upgrade

### PR DESCRIPTION
**Pull request by r-brown**

### CODE REVIEW

**Comment by r-brown** [comment]
Updated version of the [Labs64/NetLicensingClient-java](https://github.com/Labs64/NetLicensingClient-java) yet to be released.

File: pom.xml [hunk 1]
```diff
@@ -301,47 +313,29 @@
...
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-checkstyle-plugin</artifactId>
                 <version>${maven-checkstyle-plugin.version}</version>
-                <dependencies>
-                    <dependency>
-                        <groupId>com.labs64.commons</groupId>
-                        <artifactId>code-analysis</artifactId>
-                        <version>${code-analysis.version}</version>
-                    </dependency>
-                </dependencies>
```
Comment (kkorotkov) [inline pom.xml:318]: Why is this removed? Shouldn't it deliver config files for the respective plugins?

Comment (r-brown) [reply pom.xml:318]: `code-analysis` is an internal Labs64 artefact and isn't available in public.
As the Gateway module can be maintained by the external contributors, this will most likely cause a build error.

Also, delivering `SNAPSHOT` isn't the best practice.

Comment (r-brown) [reply pom.xml:318]: Probably, we need to include quality rules/metrics for the plugins in the Gateway project itself.

Comment (r-brown) [reply pom.xml:318]: Labs64 dependency replaced by code style definitions

File: pom.xml [hunk 2]
```diff
@@ -1,10 +1,11 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>com.labs64.netlicensing</groupId>
     <artifactId>gateway</artifactId>
-    <version>2.8.6-SNAPSHOT</version>
+    <version>2.8.6</version>
```
Comment (kkorotkov) [inline pom.xml]: Why switching the version to release? Master branch is for development and should stay on -SNAPSHOT. For the release, a branch to be created where the version is set to release.

**Review by kkorotkov — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (r-brown) [reply pom.xml]: `-SNAPSHOT` restored

**Review by kkorotkov — APPROVED** [review]
